### PR TITLE
Automatically generate terraform-boot API client

### PR DESCRIPTION
**Pull request by WangLiNaruto**
Automatically generate terraform-boot API client

### CODE REVIEW

**Review by swaroopar — CHANGES_REQUESTED** [review]
@WangLiNaruto i'm a little confused now. Where are we exactly using the client which YangYang generated in this?

File: modules/api/src/main/java/org/eclipse/xpanse/api/ServiceDeployV2Api.java
Comment (swaroopar) [inline modules/api/src/main/java/org/eclipse/xpanse/api/ServiceDeployV2Api.java:42]: we need to use the same existing APIs. and based on the profile.. we must be able to switch between different terraform deployment. 
Please note that even this profile based selection is also only temporary. Once we have terraform-boot fully working, we will move to that completely and delete all terraform execution within xpanse runtime. 
So its fine even if you can think of any temporary/manual solution to start the xpanse runtime to use terraform-boot for deployment

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/request/TerraformDeployRequest.java
Comment (swaroopar) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/request/TerraformDeployRequest.java:18]: i dont think we need these data classes

**Comment by WangLiNaruto** [comment]
> i'm a little confused now. Where are we exactly using the client which YangYang generated in this?

This is the code that generates the terraform-boot client. I have not yet submitted the code used in the terraform-boot client code. I will submit that code after waiting for the PR to be merged.

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**
